src/main/java/edu/ksu/ome/o365/grouper/Office365FullRefresh.java

```diff
@@ -212,14 +222,27 @@ private int deleteGroupsFromOffice365NotInGrouper(Map<String, Object> debugMap,
         return deleteCount;
     }
 
-    private Map<String, edu.internet2.middleware.grouper.changeLog.consumer.model.Group> getAllSecurityGroups(String grouperO365FolderName) {
+    private Map<String, edu.internet2.middleware.grouper.changeLog.consumer.model.Group> getAllGroups(String grouperO365FolderName) {
         GroupsOdata groupsOdata = apiClient.getAllGroups();
         Map<String, edu.internet2.middleware.grouper.changeLog.consumer.model.Group> mapToGroupName = new HashMap<>();
         for(edu.internet2.middleware.grouper.changeLog.consumer.model.Group o365Group : groupsOdata.groups){
             LOG.debug("group found is " + o365Group.displayName);
-            if(o365Group.securityEnabled && o365Group.displayName.startsWith(grouperO365FolderName)) {
-                mapToGroupName.put(o365Group.displayName, o365Group);
+            switch(apiClient.getAzureGroupType()){
+                case Unified:
+                    if(!o365Group.securityEnabled && o365Group.displayName.startsWith(grouperO365FolderName)) {
+                        mapToGroupName.put(o365Group.displayName, o365Group);
+                    }
+                    break;
+                case Security:
+                    if(o365Group.securityEnabled && o365Group.displayName.startsWith(grouperO365FolderName)) {
+                        mapToGroupName.put(o365Group.displayName, o365Group);
+                    }
+                    break;
+                case MailEnabled:
```

Comment:
Same as mentioned early... A preference to see a comment when a case statement is empty and falls through to the next line. Just makes it obvious this was intended.